### PR TITLE
Improve error messages when revealing seed phrase fails

### PR DESCRIPTION
**Pull request by jkadamczyk**
## What changed (plus any additional context for devs)

* Added a new message that is shown in case there's any issue retrieving seed. That takes care of the issue that we always displayed a message that the seed was secured with biometrics.
* Refactored the whole secret display section to be easier to reason about, now it has clear states instead of multiple booleans that then make the state
* Broke it down into smaller components
* Refactored styles to not use styled components
* Removed a state that was unreachable, that was essentially dead code

### CODE REVIEW

File: src/components/secret-display/SecretDisplaySection.tsx [hunk 1]
```diff
@@ -83,121 +97,86 @@ export default function SecretDisplaySection({
...
+            </Text>
+          </Stack>
+        </Wrapper>
+      );
+    default:
+      logger.error(
+        new RainbowError(
+          'Secret display section tires to present an unknown state'
+        )
+      );
```
Comment (jkadamczyk) [inline src/components/secret-display/SecretDisplaySection.tsx:172]: Not sure what you think about it, is there a better way to satisfy TypeScript and also make sure we are safe?

File: src/components/secret-display/SecretDisplaySection.tsx [hunk 2]
```diff
@@ -83,121 +97,86 @@ export default function SecretDisplaySection({
     // Android doesn't like to show the faceID prompt
     // while the view isn't fully visible
     // so we have to add a timeout to prevent the app from freezing
-    android
+    IS_ANDROID
```
Comment (skylarbarrera) [inline src/components/secret-display/SecretDisplaySection.tsx:100]: i wonder if a runAfterInteractions would work better here ?

**Review by skylarbarrera — APPROVED** [review]
lgtm

File: src/components/secret-display/SecretDisplaySection.tsx (hunk 1 above)
Comment (estrattonbailey) [reply src/components/secret-display/SecretDisplaySection.tsx:168]: Seems OK to me. If this ever happens we def want to know, but we don't necessarily need to blow up the UI, so returning `null` seems good to me. Any other concerns?

```suggestion
      logger.error(
        new RainbowError(
          'Secret display section tries to present an unknown state'
        )
      );
```

File: src/components/secret-display/states.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const SecretDisplayStates = {
+  loading: 'loading',
+  revealed: 'revealed',
+  noSeed: 'noSeed',
+  securedWithBiometrics: 'securedWithBiometrics',
+} as const;
+
+export type SecretDisplayStatesType = typeof SecretDisplayStates[keyof typeof SecretDisplayStates];
```
Comment (estrattonbailey) [inline src/components/secret-display/states.ts:8]: We should talk as a team about enums :P I think enum would be fine here but probably nbd

**Review by estrattonbailey — APPROVED** [review]

File: src/components/secret-display/states.ts (hunk 3 above)
Comment (jkadamczyk) [reply src/components/secret-display/states.ts:8]: I like enums, but I was trying to follow the TypeScript style guide.

**Review by ibrahimtaveras00 — APPROVED** [review]
👍🏽

**Pull request merged**